src/gui/systray.cpp

```diff
@@ -308,6 +346,15 @@ void Systray::forceWindowInit(QQuickWindow *window) const
 #endif
 }
 
+void Systray::positionNotificationWindow(QQuickWindow *window) const
+{
+    if (!useNormalWindow()) {
+        window->setScreen(currentScreen());
+        const auto position = computeNotificationPosition(window->width(), window->height());
+        window->setPosition(position);
```

Comment:
this doesn't work with wayland :( and on x doesn't follow the kde settings

Reply:
Ok, so digging into the code, the problem is this: on Linux DEs Qt's QSystemTrayIcon::geometry() is completely broken, which means we have no way of getting the position of the tray icon or, by extension, the task bar.

For the main tray window this is ok with a workaround which checks the position of the cursor and uses this as the position for the tray to open -- the idea being you'll click on the icon, thus giving the position of the icon. This breaks when trying to use it for the call notification dialog as your cursor could be anywhere on the screen at the time of the call, and the call notification position is unpredictable.

Since we have no good way of finding the taskbar position (or size) on Linux, we have to either hardcode a position for the dialog or let the window manager decide where to place it (which will usually be in the center of the screen). Hardcoding it to a corner will risk breaking the convention of the desktop in question (KDE Plasma notifies wherever the taskbar is by default, usually at the bottom-right, whereas GNOME is top-right). @jancborchardt

Reply:
For Wayland I will do some more digging

Reply:
For now I have disabled the positioning code for the notification of the system tray icon geometry is invalid

Reply:
> For now I have disabled the positioning code for the notification of the system tray icon geometry is invalid

Image for reference:

![image](https://user-images.githubusercontent.com/70155116/164911073-41926eab-e5a2-4469-b9ac-c62c6eb494d7.png)

Reply:
> we have to either hardcode a position for the dialog or let the window manager decide where to place it (which will usually be in the center of the screen)

If we have to decide between the 2, then I would say definitely hardcoding makes for a way better UX. If it’s KDE show on bottom right, and if it’s GNOME or anything else show on top right. Just use the default task bar height + some padding for vertical position, and some padding to the right side.
Sure that can still lead to overlaps in super super edge cases, but we can fix those as they are reported. Way less bad than having a notification smack dab in the middle of the screen. :)

Reply:
> > we have to either hardcode a position for the dialog or let the window manager decide where to place it (which will usually be in the center of the screen)
> 
> @claucambra If we have to decide between the 2, then I would say definitely hardcoding makes for a way better UX. If it’s KDE show on bottom right, and if it’s GNOME or anything else show on top right. Just use the default task bar height + some padding for vertical position, and some padding to the right side. Sure that can still lead to overlaps in super super edge cases, but we can fix those as they are reported. Way less bad than having a notification smack dab in the middle of the screen. :)

This is a bad idea in practice -- we do not know the size of the panels, the orientation of the panels, how DPI will affect positioning and padding of the notification, etc. A center position on the screen may not be ideal but, IMO, it is preferable to risking positioning that will be eating into people's panels or showing off-screen for users with a variety of display densities and DEs

I think the only viable solution here for notifications in corners on the Linux DEs is to default to a safe position like the center and then let the user customise the positioning through the application settings. Happy to hear if I have missed something here @mgallien @CarlSchwan

Reply:
Will this issue come up in a default installation of Ubuntu, or where would it happen?

I know hardcoding is not elegant, but putting this notification in the center is quite bad design. I'd say even if the notification is all the way in the top right / bottom right, eating into a panel, that's a better experience than in the very center.

Reply:
Wayland is now the default on GNOME (Ubuntu and Fedora) and KDE (Fedora). So in these cases the system tray and custom notification will be displayed on the middle of the screen.

Reply:
On X11 GNOME environments, the notification will now be displayed on the top-right.

For Wayland environments, as well as other DEs, we still display in center

Reply:
Yeah, so as said it sounds bad that centering this would be the default experience, even if it's only Linux. Top right and bottom right would be much better, even if it overlaps something. Cause center overlaps _everything_ always.

But cc @tobiaskaminsky on weighing the workload.